Introduce a temporary cache of variables where the place-like entities appear as constraints

jm-rivera opened the pull request:
As discussed in #41, this PR introduces a minimal implementation (with tests) of a cache of variables where the place-like entities appear as constraints.

**Note** I've only implemented this as a setting for Custom instances to keep the surface as small as possible. But it can be easily added as a setting for Base DC if needed.

keyurva commented on packages/datacommons-mcp/datacommons_mcp/clients.py:
```diff
@@ -511,7 +536,10 @@ def _filter_variables_by_existence(
+                # TODO (@jm-rivera): Remove place-like check once new search endpoint is live.
+                place_variables = self.variable_cache.get(
+                    place_dcid
+                ) | self._place_like_statvar_store.get(place_dcid, set())
```
Don't need it for this PR but wanted to mention it for the longer term: With this change, both set of places - from the variable cache and from this new store - will get returned as `places_with_data` in the response. This is technically not true for the latter in the way we model data. We may want to encode it in a separate field (say `variable_about`) so the difference is clearer and the agent can take appropriate downstream actions.

keyurva reviewed (commented):
Thanks Jorge. The overall logic looks good. I added comments on some of the details and the tests. LMK your thoughts.

jm-rivera commented on packages/datacommons-mcp/tests/test_dc_client.py:
```diff
@@ -1039,3 +1039,76 @@ def test_create_dc_client_url_computation(self, mock_dc_client):
             expected_api_url = "https://example.com/core/api/v2/"
             mock_dc_client.assert_called_with(url=expected_api_url)
+
+    @patch("datacommons_mcp.clients.DCClient._compute_place_like_statvar_store")
```
This is a very sensible idea. I've refactored various bits so that nearly all of the temp code is in a single file, and so all of the tests are in the same file. That way when we need to remove it, there will be two clear places where we need to remove code (plus a few unavoidable references in specific places)

jm-rivera commented:
Thank you for your comments @keyurva! I've refactored the code to isolate it as much as possible. And I've addressed the other tweaks/changes you requested.

keyurva commented on packages/datacommons-mcp/datacommons_mcp/_constrained_vars.py:
```diff
@@ -0,0 +1,90 @@
+from concurrent.futures import ThreadPoolExecutor, as_completed
+
+
+def _fetch_statvar_constraints(client, place_like_constraints: list[str]) -> dict:
```
Consider adding a minimal docstring similar to the one in `_extract_place_like`

keyurva replied:
Can we add a type annotation to client so it's clear which client it is.

keyurva commented on packages/datacommons-mcp/datacommons_mcp/_constrained_vars.py:
```diff
@@ -0,0 +1,90 @@
+    place_like_constraints: list[str],
+    *,
+    max_workers: int | None = None,
+) -> dict[str, list[str]]:
```
I think the return type is `dict[str, set[str]]` correct?

keyurva reviewed (approved):
Thanks for the updates and esp for isolating majority of the temp code into standalone files. This will make removing the functionality much easier when we need do.

A couple of nits but we are good to go here! Thanks again, Jorge.
